app/controllers/works_controller.rb

```diff
@@ -0,0 +1,95 @@
+class WorksController < ApplicationController
+
+    def index
+        @works = Work.all
+    end
+
+    def show
+        @work = Work.find_by(id: params[:id])
+
+        if @work.nil?
+            head :not_found
+            return
+        end 
+        @votes = @work.votes.order(created_at: :desc)
+    end
+
+    def new
+        @work = Work.new
+    end
+
+    def create
+        @work = Work.new(work_params)
+
+        if @work.save
+            flash[:message] = "Successfully created"
+            redirect_to work_path(@work)
+        else
+            flash[:error] = "Failed to create new media"
```

Comment:
To show the specifics of the failed validations, access @work.errors.messages. You can do this in your view or iterate through the messages here and store them in the `flash` hash.